grpc/diatheke.proto

```diff
@@ -28,44 +28,47 @@ service Diatheke {
   rpc Models(Empty) returns (ModelsResponse) {}
 
   // Requests a new session with the given config and returns the session
-  // ID, which is required for other rpc methods.
+  // ID, which is required for other rpc methods. After the session is
+  // created, StartSession() must be called to begin executing the Diatheke
+  // model.
   rpc NewSession(NewSessionRequest) returns (SessionID) {}
 
-  // Terminates an existing session and closes any open event streams.
+  // Begin execution of the model for the given session ID. The session's
+  // event stream should be set up prior to calling this function so that
+  // the client application can respond to any initialization events that
+  // are defined in the session's model.
+  rpc StartSession(SessionID) returns (Empty) {}
+
+  // Terminates an existing session and closes any open session streams.
   // It is an error if the SessionEndRequest has an invalid SessionID.
   rpc EndSession(SessionID) returns (Empty) {}
 
-  // Requests a new event stream for the given session.
+  // Requests a new event stream for the given session. Only one stream
+  // per session is allowed.
   rpc SessionEventStream(SessionID) returns (stream DiathekeEvent) {}
 
   // Notify Diatheke when a command has completed so that it may update
   // the dialog state. The initial command request will come as part of
-  // a DiathekeEvent. While not strictly required (depeding on the model
-  // and command), it is best practice to always call this method when a
-  // command is complete. Cases where it is required include when the 
-  // command has output parameters, or when the command is followed by 
-  // another action in the Diatheke model.
+  // a DiathekeEvent. Client applications should always call this after
+  // receiving a command event, or else the session will hang.
   rpc CommandFinished(CommandStatus) returns (Empty) {}
```

Comment:
Based on the changed comment, this was previously recommended but is now required.

> Client applications should always call this after receiving a command event, or else the session will hang.

What does this mean for any long-running commands? Are we saying that everything waits until the CommandFinished is given?

Reply:
Yes. That is exactly what it does. For long running commands, this requires an application to re-think what is considered a command. Does a command accomplish something from start to finish, or does it kick off something that runs for a long time, and once this long running thing has been started, the command is finished? Is the task itself considered the command, or is starting the task considered the command? I want to encourage the latter way of thinking. It is very hard to definitively say what state Diatheke is in otherwise.

Reply:
Should we state that explicitly, then?

Reply:
I think that kind of commentary would be more appropriate in the web documentation rather than in the obscure protobuf file that most people won't even look at.

Reply:
Or do you mean, state explicitly that Daitheke will wait until it receives the command finished notification? Making that a little more clear in the protobuf file is probably better than the phrase "the session will hang" :smile:.